pkgs/applications/networking/instant-messengers/armcord/default.nix

```diff
@@ -1,141 +1,83 @@
 
-  nativeBuildInputs = [ autoPatchelfHook dpkg makeShellWrapper wrapGAppsHook3 ];
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-UJ3E/2+MspgVbRT7s6K4lGDvTQbWD3bwyICbJjctwDI=";
+  };
 
-  dontWrapGApps = true;
+  ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
 
-  buildInputs = [
-    alsa-lib
-    at-spi2-atk
-    at-spi2-core
-    atk
-    cairo
-    cups
-    dbus
-    expat
-    ffmpeg
-    fontconfig
-    freetype
-    gdk-pixbuf
-    glib
-    gtk3
-    pango
-    systemd
-    mesa # for libgbm
-    nss
-    libuuid
-    libdrm
-    libnotify
-    libsecret
-    libpulseaudio
-    libxkbcommon
-    libappindicator-gtk3
-    xorg.libX11
-    xorg.libxcb
-    xorg.libXcomposite
-    xorg.libXcursor
-    xorg.libXdamage
-    xorg.libXext
-    xorg.libXfixes
-    xorg.libXi
-    xorg.libXrandr
-    xorg.libXrender
-    xorg.libXScrnSaver
-    xorg.libxshmfence
-    xorg.libXtst
-    wayland
-    pipewire
-  ];
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm build
 
-  sourceRoot = ".";
-  unpackCmd = "dpkg-deb -x $src .";
+    npm exec electron-builder -- \
+      --dir \
+      -c.electronDist="${electron_30}/libexec/electron" \
+      -c.electronVersion="${electron_30.version}"
+
+    runHook postBuild
+  '';
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -p "$out/bin"
-    cp -R "opt" "$out"
-    cp -R "usr/share" "$out/share"
-    chmod -R g-w "$out"
+    mkdir -p "$out/share/lib/armcord"
+    cp -r ./dist/*-unpacked/{locales,resources{,.pak}} "$out/share/lib/armcord"
 
-    # use makeShellWrapper (instead of the makeBinaryWrapper provided by wrapGAppsHook3) for proper shell variable expansion
-    # see https://github.com/NixOS/nixpkgs/issues/172583
-    makeShellWrapper $out/opt/ArmCord/armcord $out/bin/armcord \
-      "''${gappsWrapperArgs[@]}" \
-      --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}/" \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=UseOzonePlatform --enable-features=WebRTCPipeWireCapturer }}" \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
-      --suffix PATH : ${lib.makeBinPath [ xdg-utils ]}
+    install -Dm644 "build/icon.png" "$out/share/icons/hicolor/256x256/apps/armcord.png"
 
-    # Fix desktop link
-    substituteInPlace $out/share/applications/armcord.desktop \
-      --replace /opt/ArmCord/ $out/bin/
+    makeShellWrapper "${lib.getExe electron_30}" "$out/bin/armcord" \
+      --add-flags "$out/share/lib/armcord/resources/app.asar" \
+      "''${gappsWrapperArgs[@]}" \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+      --set-default ELECTRON_IS_DEV 0 \
+      --inherit-argv0
 
     runHook postInstall
   '';
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "armcord";
+      desktopName = "ArmCord";
+      exec = "${placeholder "out"}/bin/armcord %U";
```

Comment:
If possible, .desktop files shouldn't contain absolute paths, just the binary name.

I'm not able to test this ATM, but `placeholder` doesn't always behave as you'd expect when not used directly in the main derivation.
(Though, if you don't use the absolute path, you won't have to worry about it)

Reply:
This also can make DEs start the old version because there cache is not updated all the time.

Reply:
I fixed this in a new PR I made, thanks for the info :}